### ISSUE TITLE
remove the concept of relocatable files from the loader

### ISSUE THREAD

**Issue by ivg** [BinaryAnalysisPlatform/bap#1189]
Historically, we were designating a special class of binaries that were treated specially, the relocatable files. With time as we develop our backends we enabled relocation (via llvm-offset and bias) for all files, so there is no real need to separate them and we can remove a lot of code duplication.

What is more important, is that presumably due to some confusion, we output relocation and other special classes of entries only for relocatable files, despite the fact that they can also occur in regular files. So we loose important information here.

In the same venue I would like our backed not to ignore any information, as I noticed that we prune some relocations and symbols if they have zero addresses or some other values, that are not yet filled in by the system loader. This information shouldn't be hidden and, if necessary, filtered later.

Basically, I would like to see as few if/then/else as possible in the part that produces the binary specification. All the loading logic should go to the dissemblers and other downstream components. The loader should just provide information as is.

We started removing special treatment of relocatable files in #1187 but we need more work. Basically, this issue could be closed when we remove is-relocatable property from the specification.
